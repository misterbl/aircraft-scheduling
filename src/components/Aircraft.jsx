import React from "react";

const AirCraft = ({ aircraft, totalAircraftUse, onSelectAircraft }) => (
  <li onClick={onSelectAircraft} className="text-center">
    <p>{aircraft.ident}</p>
    <p>{`(${totalAircraftUse()} %)`}</p>
  </li>
);

export default AirCraft;
