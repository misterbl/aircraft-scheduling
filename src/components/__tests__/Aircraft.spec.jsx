import * as React from "react";
import { shallow } from "enzyme";
import Aircraft from "../Aircraft";

describe("Aircraft", () => {
  const props = {
    aircraft: { ident: "1234" },
    totalAircraftUse: jest.fn(() => 42)
  };
  const wrapper = shallow(<Aircraft {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
