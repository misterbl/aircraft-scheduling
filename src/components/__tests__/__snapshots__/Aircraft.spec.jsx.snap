// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aircraft matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AirCraft
    aircraft={
      Object {
        "ident": "1234",
      }
    }
    totalAircraftUse={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": 42,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          1234
        </p>,
        <p>
          (42 %)
        </p>,
      ],
      "className": "text-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1234",
        },
        "ref": null,
        "rendered": "1234",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "(42 %)",
        },
        "ref": null,
        "rendered": "(42 %)",
        "type": "p",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            1234
          </p>,
          <p>
            (42 %)
          </p>,
        ],
        "className": "text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1234",
          },
          "ref": null,
          "rendered": "1234",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "(42 %)",
          },
          "ref": null,
          "rendered": "(42 %)",
          "type": "p",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
